.github/workflows/linux-build-and-test.yaml

```diff
@@ -0,0 +1,143 @@
+name: Regression Linux
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+  pull_request:
+jobs:
+  regress_linux:
+    name: PG${{ matrix.pg }} [${{ matrix.type }}]
+    runs-on: ubuntu-18.04
+    strategy:
+
+# This is the main workflow for linux regression tests
+# Since GitHub Actions don't currently allow dynamic
+# matrix entries, the matrix has all versions but
+# for pull request we only run the configurations
+# where on_pullrequest is true
+
+      matrix:
+        pg: [ "11.0", "11.8", "12.0", "12.3" ]
+        type: [ Debug, Release, ApacheOnly ]
+
+        include:
+          - type: Debug
```

Comment:
It would be good to have some builds run on GCC and others on Clang. For instance, if we'd support Mac we could run those builds with Clang and Linux ones with GCC.